internal/app/app.go

```diff
@@ -0,0 +1,101 @@
+package app
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/Makpoc/gomigen/internal/generator"
+	"github.com/Makpoc/gomigen/internal/parser"
+	"github.com/Makpoc/gomigen/internal/version"
+)
+
+var (
+	// capitalLetterRegex captures each capital letter in a match group.
+	capitalLetterRegex = regexp.MustCompile("([A-Z])")
+)
+
+type Logger interface {
+	Printf(format string, v ...interface{})
+}
+
+type App struct {
+	pkgDir          string
+	interfaceName   string
+	outputDirectory string
+	version         string
+	logger          Logger
+}
+
+func New(pkgDir, interfaceName string, opts ...Option) *App {
+	app := &App{
+		pkgDir:        pkgDir,
+		interfaceName: interfaceName,
+		logger:        log.Default(),
+		version:       version.Version(),
+	}
+	for _, o := range opts {
+		o(app)
+	}
+	return app
+}
+
+func (a *App) Run() error {
+	middleware, err := parser.Parse(a.pkgDir, a.interfaceName)
+	if err != nil {
+		return fmt.Errorf("failed to parse interface %q: %w",
+			a.interfaceName, err)
+	}
+
+	source, err := generator.GenerateSource(middleware, a.version)
+	if err != nil {
+		return fmt.Errorf("failed to generate middleware for interface %q: %w",
+			a.interfaceName, err)
+	}
+
+	outputDir := path.Join(a.pkgDir, middleware.GenPackage)
```

Comment:
You should be using `filepath.Join` here as well.

Reply:
(as well as the lines below)

Reply:
I should have.. I am now